### DISCUSSION TITLE
Wazuh IIS issue

### DISCUSSION THREAD

**Discussion by rajkshk2** [wazuh/wazuh#26594] [Q&A] [answered]
Hi Team I am fcaing below issue while trying to generate alert in wazuh:
**Phase 1: Completed pre-decoding.
	full event: '{"timestamp":"2024-10-29T03:22:09.724+0000","agent":{"id":"001","name":"ATTACKER","ip":"192.168.50.131"},"manager":{"name":"wazuh-server"},"id":"1730172129.288489","full_log":"2024-10-29 03:21:42 192.168.50.131 GET / - 80 - 192.168.50.1 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/130.0.0.0+Safari/537.36 - 304 0 0 43","decoder":{"parent":"windows-date-format","name":"web-accesslog-iis-default"},"data":{"action":"GET","srcip":"192.168.50.1","srcport":"80","id":"304","url":"/ -","user_agent":"Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36 (KHTML,+like+Gecko)+Chrome/130.0.0.0+Safari/537.36"},"location":"C:\\inetpub\\logs\\LogFiles\\W3SVC1\\u_ex241029.log"}'

**Phase 2: Completed decoding.
	name: 'json'
	agent.id: '001'
	agent.ip: '192.168.50.131'
	agent.name: 'ATTACKER'
	data.action: 'GET'
	data.id: '304'
	data.srcip: '192.168.50.1'
	data.srcport: '80'
	data.url: '/ -'
	data.user_agent: 'Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36 (KHTML,+like+Gecko)+Chrome/130.0.0.0+Safari/537.36'
	decoder.name: 'web-accesslog-iis-default'
	decoder.parent: 'windows-date-format'
	full_log: '2024-10-29 03:21:42 192.168.50.131 GET / - 80 - 192.168.50.1 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/130.0.0.0+Safari/537.36 - 304 0 0 43'
	id: '1730172129.288489'
	location: 'C:\inetpub\logs\LogFiles\W3SVC1\u_ex241029.log'
	manager.name: 'wazuh-server'
	timestamp: '2024-10-29T03:22:09.724+0000'

**Phase 3: Completed filtering (rules).
	id: '1002'
	level: '2'
	****description: 'Unknown problem somewhere in the system.'****
	groups: '["syslog","errors"]'
	firedtimes: '3'
	gpg13: '["4.3"]'
	mail: 'false'
**Alert to be generated.

**Comment by Hasitha9796** [comment, marked as answer]
Hi [rajkshk2](https://github.com/rajkshk2)

You did not test the logs correctly. When capturing logs from archives.json, you need to copy the `full_log` content from the archived log entry.

We recommend creating custom rules and decoders based on archives.json because these logs contain the `full_log` field, which is the actual data being parsed by the analysis engine.
Below is an example of how a typical `archives.json` event should look (the required part is highlighted separately):
```
{"timestamp":"2023-09-05T02:47:40.074+0000","agent":{"id":"001","name":"abc","ip":"10.0.2.29},"manager":{"name":"Server85"},"id":"1693882060.373586","full_log ":"Sep 5 03:10:19 Server91 dbus-daemon[676]: [system] Successfully activated service 'org.freedesktop.UPower","predecoder":{"program_name":"dbus-daemon","timestamp":"Sep 5 03:10:19","hostname":"Server91"},"decoder":{},"location":"/var/log/syslog"}
```
From that archives.json log, you need to test `full_log` part.
For example:

`Sep 5 03:10:19 Server91 dbus-daemon[676]: [system] Successfully activated service 'org.freedesktop.UPower`
Ref: https://wazuh.com/blog/creating-decoders-and-rules-from-scratch/

Enable archives.json log, set the `<logall_json>yes</logall_json>` to yes at `/var/ossec/etc/ossec.conf` file of the Wazuh manager.
Documentation:[Wazuh Documentation | logall](https://documentation.wazuh.com/current/user-manual/manager/event-logging.html#enabling-archiving)
```
<ossec_config>

 <global>

___________________

  <logall_json>yes</logall_json>

_______________
```
This option will allow you to see all the events being monitored by your manager in the /var/ossec/logs/archives/archives.json file. You will then be able to observe the incoming logs generated by your endpoints. After setting this option, restart the manager and check the archives.json file.
Note: Don't forget to disable the logall parameter once you have finished troubleshooting. Leaving it enabled could lead to high disk space consumption.
Look for if there are any logs inside the archive log which is relevant. Use grep parameters related to the log.
`cat /var/ossec/logs/archives/archives.json | grep Keywoard`

According to your log, you need to test this part.
```
"2024-10-29 03:21:42 192.168.50.131 GET / - 80 - 192.168.50.1 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/130.0.0.0+Safari/537.36 - 304 0 0 43","decoder":{"parent":"windows-date-format","name":"web-accesslog-iis-default"},"data":{"action":"GET","srcip":"192.168.50.1","srcport":"80","id":"304","url":"/ -","user_agent":"Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36 (KHTML,+like+Gecko)+Chrome/130.0.0.0+Safari/537.36"}
```
Ref: https://documentation.wazuh.com/current/user-manual/ruleset/testing.html

I tested the log and confirmed that it matches rule ID `31108` (Ignored URLs - simple queries). However, this rule has a level of 0, which is why it doesn't appear on the dashboard. By default, Wazuh only displays alerts with a level of 3 or higher on the dashboard.

If you need to overwrite this existing rule to level 3 and need this alert on the dashboard, you can refer to this guide.
Ref: https://documentation.wazuh.com/current/user-manual/ruleset/rules/custom.html#changing-existing-rules

Let me know if you need further assistance on this.
